ui/packages/consul-ui/app/abilities/partition.js

```diff
@@ -12,15 +13,24 @@ export default class PartitionAbility extends BaseAbility {
   }
 
   get canManage() {
-    return this.canCreate;
+    // management currently means "can I write", not necessarily just create
+    return this.canWrite;
+  }
+
+  get canCreate() {
+    // we can only currently create a partition if you have only one datacenter
+    if (this.dcs.peekAll().length > 1) {
+      return false;
+    }
+    return super.canCreate;
   }
 
   get canDelete() {
     return this.item.Name !== 'default' && super.canDelete;
   }
 
   get canChoose() {
-    if(typeof this.dc === 'undefined') {
+    if (typeof this.dc === 'undefined') {
```

Comment:
Can we do better?  [For example, Ember `isEmpty()`](https://api.emberjs.com/ember/release/functions/@ember%2Futils/isEmpty)?

Reply:
We don't use this Ember type things anywhere in the codebase at all, and prefer to use vanilla JS equivalents, and we use this specific form everywhere. I've found to be "caught out" by these Ember things not doing what you would expect sometimes. There's comment somewhere else recently similar to this, will find in a sec.

Reply:
Totally get the desire to go vanilla.  Can it at least be wrapped in a custom utility function that you own?  `isUndefined()` for instance, to clean things up a little.  Just as a rule of thumb I try to avoid typeof and direct string matching in core code.  Some types are complicated to test, for example arrays and NaN, and really _need_ helpers.  All of this is potentially an argument for typed JS.  In the absence of that though, the logic could be wrapped.

Reply:
I'm strongly against changing this.

> Some types are complicated to test, for example arrays and NaN

We wouldn't use this form for Arrays or NaN as there are different native vanilla methods for type checking those which we'd use instead of a `typeof` (or instead of a custom helper function).

Folks quite often change the definition of "Undefined-ness" (is it empty, or undefined, or is it just null, or maybe {size: 0}) Ember functions being a case in point, which just leads to confusion. If we were to have `isUndefined` then we'd also need `isFunction` and then we get on another slippery slope (do we need an isClass and isMethod?), which could also be misinterpreted. In fact I find it quite relieving to find a codebase using these vanilla approaches, as I know I'm not going to be caught out by some re-interpretation of undefined or empty (I believe a lot of Ember things also use these under the hood instead of _their own_ helper functions)

If we did it here, we should change the entire codebase to use a helper function, lots of churn for not a lot of benefit.

Given the fact that any usage of a custom function would need a corresponding import, it's also easier to shorter to type the way it currently is.

I think given we are only talking about this due to prettier (a tool to avoid bikeshedding) adding a space here, its all a bit 🤷 😆 

Honestly, I can't think of many benefits to using a helper function.

Reply:
It's an interesting conversation nonetheless.  Thanks.